server.go

```diff
@@ -186,5 +214,87 @@ type ServerTripper struct{}
 
 func (t *ServerTripper) RoundTrip(req *http.Request) (res *http.Response, err error) {
 	req.Header.Set("User-Agent", serverUserAgent)
+
+	//TODO: Move this whole block elsewhere
```

Comment:
> Move this whole block elsewhere
I'll take care of that. Mind explaining what this part exactly does?

Reply:
It's for the hmac auth.
The method needs either the query string or the whole body to generate the `X-Signature` header required for authentication.
Auth is not necessary when the 'Worker' has not been created yet